### PR TITLE
Prevent podcast and episode theft

### PR DESCRIPTION
**Pull request by chrisrhoden**

### CODE REVIEW

File: app/policies/episode_policy.rb [hunk 1]
```diff
@@ -4,15 +4,24 @@ def create?
...
 
+  private
+
   def account_id
     resource.podcast.account_id
   end
+
+  def account_id_was
+    if resource.podcast_id_changed? && resource.podcast_id_was.present?
+      Podcast.find(resource.podcast_id_was)
```
Comment (cavis) [inline app/policies/episode_policy.rb:22]: Ah, so also catching the case where they `set_podcast_uri`

File: app/policies/application_policy.rb [hunk 2]
```diff
@@ -10,4 +10,23 @@ def update?
...
+      token&.authorized?(account_id, scope)
+    end
+  end
+
+  def account_id
+    resource.account_id
+  end
+
+  def account_id_was
+    nil
```
Comment (cavis) [inline app/policies/application_policy.rb:30]: Why nil?  (vs `resource.account_id_was`)

Comment (chrisrhoden) [reply app/policies/application_policy.rb:30]: Could go either way, I guess I was imagining that we shouldn't do this check by default but :shrug:

File: app/policies/episode_policy.rb (hunk 1 above)
Comment (chrisrhoden) [reply app/policies/episode_policy.rb:22]: Yes - I actually think that account_id_was is unnecessary given the existing controllers but completeness doesn't hurt in this case.

**Pull request merged**